sia/core.go

```diff
@@ -205,6 +205,7 @@ func (c *Core) ScanMutexes() {
 
 // Close does any finishing maintenence before the environment can be garbage
 // collected. Right now that just means closing the server.
+// TODO: save wallet
```

Comment:
ah. In that case I think that the "Write & Copy" approach is good enough. Unless handled carefully, triggering a save right before exit might even lead to more frequent corruption, not less.